application/src/main/java/org/togetherjava/tjbot/commands/free/FreeUtil.java

```diff
@@ -0,0 +1,95 @@
+package org.togetherjava.tjbot.commands.free;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.utils.TimeUtil;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.config.FreeCommandConfig;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * A class containing helper methods required by the free package command.
+ */
+enum FreeUtil {
+    ;
+    private static final Logger logger = LoggerFactory.getLogger(FreeUtil.class);
+
+    /**
+     * Helper method to easily send ephemeral messages to users.
+     * 
+     * @param interaction The event or hook that this message is responding to
+     * @param message The text to be display for the user to read.
+     */
+    public static void sendErrorMessage(@NotNull Interaction interaction, @NotNull String message) {
+        interaction.reply(message).setEphemeral(true).queue();
+    }
+
+    /**
+     * Method that provides the message history of a {@link TextChannel}.
+     * <p>
+     * </p>
+     * This method attempts to retrieve the message history, and logs any problems that occur in the
+     * attempt.
+     *
+     * @param channel the channel from which the history is required.
+     * @param limit the number of messages to retrieve.
+     * @return the requested message history or empty if unable to.
+     */
+    public static @NotNull Optional<List<Message>> getChannelHistory(@NotNull TextChannel channel,
+            final int limit) {
+        return Optional
+            .ofNullable(channel.getHistory().retrievePast(limit).mapToResult().map(listResult -> {
+                if (listResult.isFailure()) {
+                    logger.error("Failed to retrieve messages from %s because of:"
+                        .formatted(channel.getAsMention()), listResult.getFailure());
+                    return null;
+                } else {
+                    return listResult.get();
+                }
+            }).complete());
```

Comment:
I think this would be easier to read without the all-wrapping `Optional.ofNullable`. So:
```java
return channel.getHistory().retrievePast(limit).mapToResult().map(listResult -> {
  if (...) {
    logger.error(...);
    return Optional.empty();
  }
  return Optional.of(listResult.get());
}).complete();
```

Reply:
fixed